tests/session/test_cdp.py

```diff
@@ -158,3 +159,52 @@ async def test_cdp_wait_for_event(websocket, get_cdp_session_id, context_id):
             "session": session_id
         }
     })
+
+
+@pytest.mark.asyncio
+async def test_cdp_no_extraneous_events(websocket, get_cdp_session_id,
+                                        create_context, url_base):
+    new_context_id = await create_context()
+    await execute_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": url_base,
+                "wait": "complete",
+                "context": new_context_id
+            }
+        })
+
+    await subscribe(websocket, ["cdp"], [new_context_id])
+
+    session_id = await get_cdp_session_id(new_context_id)
+
+    id = await send_JSON_command(
+        websocket, {
+            "method": "cdp.sendCommand",
+            "params": {
+                "method": "Target.attachToTarget",
+                "params": {
+                    "targetId": new_context_id,
+                },
+                "session": session_id
+            }
+        })
+
+    events = []
+    event = await read_JSON_message(websocket)
+
+    session_id = None
+    with pytest.raises(asyncio.TimeoutError):
+        while True:
+            if 'id' in event and event['id'] == id:
+                session_id = event['result']['result']['sessionId']
+            if 'id' not in event:
+                events.append(event)
+            event = await asyncio.wait_for(read_JSON_message(websocket),
+                                           timeout=1.0)
```

Comment:
Can you send another command an verify it is finished with no extra messages instead of waiting for 1 second?

Reply:
Could you post a suggestion in code? I didn't get the idea

Reply:
https://github.com/GoogleChromeLabs/chromium-bidi/commit/e864c4cfb59057ad29533e787df19c9885c810d1

Reply:
The timeout was not the cause for the issue, I have updated the PR to reflect the fix.